lib/actions/receive-message.test.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+
+const dedent = require('dedent');
+const { receiveMessage } = require('../sqs');
+const actionReceiveMessage = require('./receive-message');
+
+jest.mock('../sqs');
+
+describe('actions/receive-message', () => {
+  test('throw error when receiveMessage throws error', () => {
+    receiveMessage.mockImplementationOnce(() => {
+      throw Object.assign(new Error('foo'), { code: 'bar' });
+    });
+
+    const expectedResponse = dedent`<?xml version="1.0"?>
+    <ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/">
+      <Error>
+        <Code>bar</Code>
+        <Detail/>
+        <Message>foo</Message>
+        <Type>Sender</Type>
+      </Error>
+      <RequestId>00000000-0000-0000-0000-000000000000</RequestId>
+    </ErrorResponse>`;
+
+    let error;
```

Comment:
Possibly a cleaner version:

```js
const expectedError = Object.assign(new Error('foo'), { code: 'bar', xml: expectedResponse });
expect(() => actionReceiveMessage({ QueueUrl: 'foo-bar' })).toThrow(expectedError);
```

Reply:
This won't work, since expectedError does not get asserted correctly